core/src/main/java/io/questdb/cutlass/line/Sender.java

```diff
@@ -0,0 +1,507 @@
+ * <br>
+ * Thread-safety: Sender is not thread-safe. Each thread-safe needs its own instance or you have to implement
+ * a mechanism for passing Sender instances among thread. An object pool could have this role.
+ * <br>
+ * Error-handling: Most errors throw an instance of {@link LineSenderException}.
+ *
+ */
+public interface Sender extends Closeable {
+
+    /**
+     * Select the table for a new row. This is always the first method to start a error. It's an error to call other
+     * methods without calling this method first.
+     *
+     * @param table name of the table
+     * @return this instance for method chaining
+     */
+    Sender table(CharSequence table);
+
+    /**
+     * Add a column with an integer value.
+     *
+     * @param name name of the column
+     * @param value value to add
+     * @return this instance for method chaining
+     */
+    Sender longColumn(CharSequence name, long value);
+
+    /**
+     * Add a column with a string value.
+     *
+     * @param name name of the column
+     * @param value value to add
+     * @return this instance for method chaining
+     */
+    Sender stringColumn(CharSequence name, CharSequence value);
+
+    /**
+     * Add a column with a floating point value.
+     *
+     * @param name name of the column
+     * @param value value to add
+     * @return this instance for method chaining
+     */
+    Sender doubleColumn(CharSequence name, double value);
+
+    /**
+     * Add a column with a boolean value.
+     *
+     * @param name name of the column
+     * @param value value to add
+     * @return this instance for method chaining
+     */
+    Sender boolColumn(CharSequence name, boolean value);
+
+    /**
+     * Add a column with a non-designated timestamp value.
+     *
+     *
+     * @param name name of the column
+     * @param value value to add (in microseconds since epoch)
+     * @return this instance for method chaining
+     */
+    Sender timestampColumn(CharSequence name, long value);
+
+    /**
+     * Add a column with a symbol value. You must call add symbols before adding any other column types.
+     *
+     * @param name name of the column
+     * @param value value to add
+     * @return this instance for method chaining
+     */
+    Sender symbol(CharSequence name, CharSequence value);
+
+    /**
+     * Finalize the current row and let QuestDB server to assign a timestamp. If you need to set timestamp
+     * explicitly then see {@link #atMicros(long)}.
+     * <br>
+     * After calling this method you can start a new row by calling {@link #table(CharSequence)} again.
+     *
+     */
+    void atNow();
+
+    /**
+     *  Finalize the current row and assign an explicit timestamp in microseconds since Epoch.
+     *  After calling this method you can start a new row by calling {@link #table(CharSequence)} again.
+     *
+     * @param timestamp timestamp in Epoch nanoseconds.
+     */
+    void atMicros(long timestamp);
```

Comment:
That's a good question! Here are the reasons I picked micros:
- ordinary timestamps columns use microseconds too 
- it's a native unit used by questdb
- I see it as a sensible compromise between coarse-grained milliseconds and overly small nanoseconds. 

If you have timestamps in millis then it's trivial to convert them to microseconds. If that's proven to be error-prone then we could add `atMillis()`. We could offer even more high-level type such as `java.time.Instant` which would eliminate the ambiguity completely. But I didn't want to do this in a first release because I believe micro is sufficient. Adding a new method is simple. Removing is not. 

And why not nanos? QuestDB uses microseconds internally so there is not much to gain by using microseconds at the API level. At the same time, I believe nanoseconds are too abstract for many developers. 

Let me know if you think otherwise.

Reply:
I feel `atMillis()` will be more appropriate, some users send mills and hardly know what micros are. I would simplify API for them.